src/Exporter/DataSourceInterface.php

```diff
@@ -20,6 +20,8 @@ interface DataSourceInterface
 {
     /**
      * @param string[] $fields
+     *
+     * @return SourceIteratorInterface<mixed>
```

Comment:
Can we be more specific? I guess it should return a `SourceIteratorInterface` for arrays.

Reply:
I don't really know.

Here it returns 
https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/29aeea8ba292ac3f626c56e78484b2c10b04cbe2/src/Exporter/DataSource.php#L64

And indeed, it's an array
https://github.com/sonata-project/exporter/blob/2.x/src/Source/AbstractPropertySourceIterator.php#L131

BTW I should also update AdminInterface::getDataSourceIterator

But since the exporter bundle has the following code 
```
    foreach ($this->source as $data) {
        $this->writer->write($data);
    }
```
And write is requiring an array, so the template
https://github.com/sonata-project/exporter/blob/2.x/src/Source/SourceIteratorInterface.php#L17
Could be improved to T of array.

Reply:
https://github.com/sonata-project/exporter/pull/465